### PR TITLE
Fix pair implementation issues

### PR DESCRIPTION
**Pull request by PointKernel**
- Fixes a bug of std pair traits bug unveiled by cudf [CI](https://github.com/rapidsai/cudf/actions/runs/5469754572/jobs/9959101572?pr=13665)
- Removes the redundant file

### CODE REVIEW

File: include/cuco/detail/pair.cuh [hunk 1]
```diff
@@ -1,295 +0,0 @@
...
- *
- * @tparam pair_type The pair type that will be packed
- *
- * @return true If the pair type can be packed
- * @return false  If the pair type cannot be packed
- */
-template <typename pair_type,
-          typename key_type   = typename pair_type::first_type,
-          typename value_type = typename pair_type::second_type>
-constexpr bool is_packable()
```
Comment (sleeepyjack) [inline include/cuco/detail/pair.cuh:132]: Do we no longer need this helper type?

File: include/cuco/detail/pair.cuh [hunk 2]
```diff
@@ -1,295 +0,0 @@
...
-   */
-  __device__ pair_converter(packed_type p) : packed{p} {}
-};
-
-}  // namespace detail
-
-/**
- * @brief Custom pair type
- *
- * This is necessary because `thrust::pair` is under aligned.
```
Comment (sleeepyjack) [inline include/cuco/detail/pair.cuh:175]: Has this been fixed or why can we remove this now?

**Review by sleeepyjack — COMMENTED** [review]
Thanks! Let's discuss this PR in our sync.

File: include/cuco/detail/pair.cuh (hunk 1 above)
Comment (sleeepyjack) [reply include/cuco/detail/pair.cuh:132]: Resolved due to offline discussion

**Review by sleeepyjack — APPROVED** [review]

File: include/cuco/detail/pair.cuh (hunk 2 above)
Comment (PointKernel) [reply include/cuco/detail/pair.cuh:175]: Moved to `detail/utils.cuh`

**Pull request merged**